SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -100,30 +100,45 @@ public async Task<DepotManifest> DownloadManifestAsync( uint depotId, ulong mani
                     throw new SteamKitWebRequestException( $"Response status code does not indicate success: {response.StatusCode:D} ({response.ReasonPhrase}).", response );
                 }
 
-                if ( !response.Content.Headers.ContentLength.HasValue )
-                {
-                    throw new SteamKitWebRequestException( "Response does not have Content-Length", response );
-                }
-
                 cts.CancelAfter( ResponseBodyTimeout );
 
-                var contentLength = ( int )response.Content.Headers.ContentLength;
-                var buffer = ArrayPool<byte>.Shared.Rent( contentLength );
+                var contentLength = -1;
+                byte[]? buffer = null;
+
+                if ( response.Content.Headers.ContentLength.HasValue )
+                {
+                    contentLength = ( int )response.Content.Headers.ContentLength;
+                    buffer = ArrayPool<byte>.Shared.Rent( contentLength );
+                }
+                else
+                {
+                    DebugLog.WriteLine( nameof( CDN ), $"Manifest response does not have Content-Length, falling back to unbuffered read." );
+                }
 
                 try
                 {
-                    using var ms = new MemoryStream( buffer, 0, contentLength );
+                    MemoryStream ms;
 
-                    // Stream the http response into the rented buffer
-                    await response.Content.CopyToAsync( ms, cts.Token );
+                    if ( buffer != null )
+                    {
+                        ms = new MemoryStream( buffer, 0, contentLength );
 
-                    if ( ms.Position != contentLength )
+                        // Stream the http response into the rented buffer
+                        await response.Content.CopyToAsync( ms, cts.Token );
+
+                        if ( ms.Position != contentLength )
+                        {
+                            throw new InvalidDataException( $"Length mismatch after downloading depot manifest! (was {ms.Position}, but should be {contentLength})" );
+                        }
+
+                        ms.Position = 0;
+                    }
+                    else
                     {
-                        throw new InvalidDataException( $"Length mismatch after downloading depot manifest! (was {ms.Position}, but should be {contentLength})" );
+                        var data = await response.Content.ReadAsByteArrayAsync();
```

Comment:
Are we avoiding `ReadAsStreamAsync` because we need to be able to seek?

Reply:
ZipArchive(Stream) is a lie because it will copy the stream into an internal MemoryStream if stream is not seekable: https://source.dot.net/#System.IO.Compression/System/IO/Compression/ZipArchive.cs,138

Reply:
lol